src/tests/basic/combat.ts

```diff
@@ -0,0 +1,200 @@
+import { GameState, PlayerState } from "../../framework/types";
+import { P1, P2, initDummyGameState } from "../testhelper";
+import {
+  getAttack,
+  getHealth,
+  getPossibleAttackTargets,
+} from "../../framework/queries/combat";
+
+import { expect } from "chai";
+import { attackInstance } from "../../framework/actions/combat";
+
+const addInstance = (
+  $: GameState,
+  id: string,
```

Comment:
nit: It may be simpler / cleaner if instead of passing a desired iid in we generate it dynamically (the same way the actual game does) and return it